Tests/MongoSwiftTests/AsyncAwaitTestUtils.swift

```diff
@@ -0,0 +1,137 @@
+#if compiler(>=5.5) && canImport(_Concurrency) && os(Linux)
+
+import Foundation
+import MongoSwift
+import NIO
+import TestsCommon
+import XCTest
+
+/// Temporary utility function until XCTest supports `async` tests.
+func testAsync(_ block: @escaping () async throws -> Void) {
+    let group = DispatchGroup()
+    group.enter()
+    Task.detached {
+        do {
```

Comment:
I realized I needed to add a do/catch here, because otherwise if the call to `block` threw an error the Task would finish without ever calling `group.leave()` and so the tests would hang indefinitely.